src/main/java/org/folio/rest/impl/PurchaseOrderLineHelper.java

```diff
@@ -168,30 +166,84 @@ private CompositePurchaseOrder validateOrderState(CompositePurchaseOrder po) {
     }
   }
 
+  public static boolean isCreateInventoryNull(CompositePoLine compPOL) {
+    if (null == compPOL.getOrderFormat()) {
+      return false;
+    }
+    switch (compPOL.getOrderFormat()) {
+    case P_E_MIX:
+      return isEresourceInventoryNotPresent(compPOL)
+          || isPhysicalInventoryNotPresent(compPOL);
+    case ELECTRONIC_RESOURCE:
+      return isEresourceInventoryNotPresent(compPOL);
+    case OTHER:
+    case PHYSICAL_RESOURCE:
+      return isPhysicalInventoryNotPresent(compPOL);
+    default:
+      return false;
+    }
+  }
+
+  private static Boolean isPhysicalInventoryNotPresent(CompositePoLine compPOL) {
+    return Optional.ofNullable(compPOL.getPhysical())
+      .map(physical -> physical.getCreateInventory() == null)
+      .orElse(true);
+  }
+
+  private static Boolean isEresourceInventoryNotPresent(CompositePoLine compPOL) {
+    return Optional.ofNullable(compPOL.getEresource())
+      .map(eresource -> eresource.getCreateInventory() == null)
+      .orElse(true);
+  }
+
+  /**
+   * get the tenant configuration for the orderFormat, if not present assign the defaults
+   * Default values:
+   * Physical : CreateInventory.INSTANCE_HOLDING_ITEM
+   * Eresource: CreateInventory.INSTANCE_HOLDING
+   *
+   * @param compPOL
+   * @param jsonConfig
+   */
   private void updateCreateInventory(CompositePoLine compPOL, JsonObject jsonConfig) {
-    // try to set createInventory by values from mod-configuration. If empty - set default hardcoded values
-    if (compPOL.getEresource() != null && compPOL.getEresource().getCreateInventory() == null) {
+    // try to set createInventory by values from mod-configuration. If empty -
+    // set default hardcoded values
+    if (compPOL.getOrderFormat().equals(OrderFormat.ELECTRONIC_RESOURCE)
+        || compPOL.getOrderFormat().equals(OrderFormat.P_E_MIX)) {
       String tenantDefault = jsonConfig.getString(ERESOURCE);
-      Eresource.CreateInventory eresourceDefaultValue = StringUtils.isEmpty(tenantDefault) ?
-        Eresource.CreateInventory.INSTANCE_HOLDING : Eresource.CreateInventory.fromValue(tenantDefault);
-      compPOL.getEresource().setCreateInventory(eresourceDefaultValue);
+      Eresource.CreateInventory eresourceDefaultValue = getEresourceInventoryDefault(tenantDefault);
+      if (compPOL.getEresource() == null) {
+        compPOL.setEresource(new Eresource());
+      }
+      if(isEresourceInventoryNotPresent(compPOL)) {
```

Comment:
Optional. This condition might be part of OrderFormat check above as it may turn out that no need to set value as it is already set...

Reply:
This is to cover the case of a P/E mix order Format, and an eresource block is given with a createInventory block and a physical is blank, we are supposed to set the default only for the Physical block
for ex:
{
"orderFormat":"P/E Mix"
"eresource":{
"createInventory":"Instance"
}

Reply:
As I mentioned this is optional change...
By the way I think for the sample above `isEresourceInventoryNotPresent` will return `false` as the method checks only `eresource` section...

Reply:
correct, that is the expected behavior because we don't want to set the createInventory for eresource in this case, and set it only for physical